littleblog/entry/urls.py

```diff
@@ -3,9 +3,9 @@
 from .views import UserProfileCreateView, EntryDetailView, RateUpdateView, EntryCreateView
 
 urlpatterns = [
-    path('CreateUserProfile/', UserProfileCreateView.as_view()),
-    path('EntryDetail/<int:pk>/', EntryDetailView.as_view()),
-    path('EntryCreate/', EntryCreateView.as_view()),
-    path('set_mark/', RateUpdateView.as_view())
+    path('create_profile/', UserProfileCreateView.as_view()),
+    path('entries/<int:pk>/', EntryDetailView.as_view()),
+    path('entries/create/', EntryCreateView.as_view()),
```

Comment:
By REST guidelines we should use http methods for separation of different actions on API resource.

There's some formal standart of RESTful API structure established. It's easier to explain on example, see this:

GET /users/  - list of all users;
POST /users/  - create user action;
GET /users/<pk>/  - specific user details;
PUT/PATCH /users/<pk>/  - update user details;
DELETE /users/<pk>/  - delete user. 

Follow that in your application and you won't be wrong.

Reply:
So, based on this requirements, I need to combine each model's CRUD operations in one View? There is no another way to differ CRUD actions only based on http method with the same endpoints names?

Reply:
Not really. If you're using `View` approach you match one URL template with one `View` class. `/users/` and `/users/<pk>/` are different URLs and they should be connected to separate `View`s.

So, for listing all users and creating a new user you would use one View and for operations with specific user you would use another View. 

```
UsersListCreateView(generics.ListCreateAPIView):
    queryset ...
    serializer_class ...

    def perform_create():
        ....
```

Note that its possible to use separate different serializers for different actions. Example:
```
def get_serializer_class(self):
        if self.request.method == 'GET':
            return UserListSerializer
        elif self.request.method == 'POST':
            return UserCreateSerializer
```

Reply:
You can jump ahead and try to implement ViewSet. ViewSet is actually the thing that allows to write one class for one resource, covering all operation (CRUD + custom). https://www.django-rest-framework.org/api-guide/viewsets/

But I'd suggest to move to viewsets in further iterations / in separate PR.